specs/hierarchical_consensus.md

```diff
@@ -0,0 +1,1079 @@
+[Path messages](#Path-messages "Path messages") | Reliable | No | |__FIP__|
+[Subnet Content Resolution Protocol](#Subnet-Content-Resolution-Protocol "Subnet Content Resolution Protocol") | Reliable | No | |__FIP__ |
+[Resolution approaches](#Resolution-approaches "Resolution approaches") | Reliable | No | |__FIP__ |
+[Data availability](#Data-availability "Data availability") | WIP/Draft |  | |N/A |
+[Atomic Execution Protocol](#Atomic-Execution-Protocol "Atomic Execution Protocol") | Reliable | No | |__FIP__|
+[Collateral and slashing](#Collateral-and-slashing "Collateral and slashing") | WIP/Draft |  | |N/A |
+[Detectable misbehaviors](#Detectable-misbehaviors "Detectable misbehaviors") | WIP/Draft |   | |N/A |
+[Cross-net routing gas price](#Cross-net-routing-gas-price "Cross-net routing gas price") | WIP/Draft |   | |N/A |
+
+## Introduction
+At a high level, hierarchical consensus (HC) allows for incremental, on-demand blockchain scaling and simplifies the deployment of new use cases with clearly isolated security domains that provide flexibility for varied use cases.
+
+Consensus, or establishing total order across transactions, poses a major scalability bottleneck in blockchain networks. This is particularly the case when all validators are required to process all transactions. Regardless of the specific consensus protocol implementation used, this makes blockchains unable to increase their performance by adding more participants (scale-out).
+
+In traditional distributed computing, one possible approach to overcoming this limitation is to resort to the partitioning, or sharding, of state processing and transaction ordering. In a sharded system the blockchain stack is divided into different groups called shards, each operated by its own set of nodes, which keep a subset of the state and are responsible for processing a part of the transactions sent to the system. 
+
+The main challenge with applying traditional sharding to the Byzantine fault-tolerant context of the blockchain lies in the security/performance tradeoff. As miners are assigned to shards, there is a danger of diluting security when compared to the original single-chain (single-shard) solution. In both proof-of-work and proof-of-stake (PoS) blockchains, sharding may give the attacker the ability to compromise a single shard with only a  fraction of the mining power, potentially compromising the system as a whole. Such attacks are often referred to as _1\% attacks_. To circumvent them, sharding systems need to periodically reassign miners to shards in an unpredictable way, so as to cope with a semi-dynamic adversary. We believe that this traditional approach to scaling, which considers the system as a monolith, is not suitable for decentralized blockchains due to their complexity.  <!--and the fact that sharded systems reshuffle state without the consent of its owners, disrupting use cases that benefit from finer control over the system topology.-->
```

Comment:
"Such attacks are often referred to as 1% attacks. " --- perhaps give reference(s)